server/utils/controllerUtils.ts

```diff
@@ -30,26 +29,16 @@ export default class ControllerUtils {
     req: Request,
     res: Response,
     draftsService: DraftsService,
-    {
-      idParamName,
-      notFoundUserMessage,
-      typeName,
-    }: { idParamName: string; notFoundUserMessage: string; typeName: string }
+    { idParamName }: { idParamName: string; notFoundUserMessage: string; typeName: string }
   ): Promise<DraftFetchResult<T>> {
     const id = req.params[idParamName]
     const draft = await draftsService.fetchDraft<T>(id, {
       userId: res.locals.user.userId,
     })
 
-    if (draft === null) {
-      throw createError(500, `UI-only draft ${typeName} with ID ${id} not found`, {
-        userMessage: notFoundUserMessage,
-      })
-    }
-
-    if (draft.softDeleted) {
+    if (draft === null || draft.softDeleted) {
       res.status(410 /* Gone */)
-      const view = new DraftSoftDeletedView()
+      const view = new DraftSoftDeletedOrNotFoundView()
```

Comment:
_semantics_
Along with the notion of "intertwined things increase semantic complexity", would it be worth having separate views for the two sides? Expired/deleted/missing might have separate needs

Reply:
Good point, I've split it into two views